### ISSUE TITLE
Set default SpEL to configure partition key expression for outbound messages

### ISSUE THREAD

**Issue by ilayaperumalg** [spring-cloud/spring-cloud-stream#1040]
_From @ashvinkanani-dream on August 17, 2017 9:9_

Hi,

I'd like to configure the partition key expression for all outbound messages being sent by any producer.

To achieve this I have configured following property
```
spring.cloud.stream.default.producer.partitionKeyExpression=payload.id
```

But, it didn't worked for me.
Also, when I configured the property for each output channel, it worked fine.

```
spring.cloud.stream.bindings.output-channel-1.producer.partitionKeyExpression=payload.id
spring.cloud.stream.bindings.output-channel-2.producer.partitionKeyExpression=payload.id
```

Reference
- [Producer Properties](http://docs.spring.io/spring-cloud-stream/docs/current/reference/htmlsingle/#_producer_properties)

_Copied from original issue: spring-cloud/spring-cloud-stream-binder-kafka#182_

**Comment by ilayaperumalg** [comment]
Hi @ashvinkanani-dream 

Thanks for reporting this.

It seems like the `conversion` (String to Expression) doesn't take place when binding the default producer's `partitionKeyExpression` into `BindingProperties` [here](https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/EnvironmentEntryInitializingTreeMap.java#L77)

**Comment by xuben** [comment]
Hi, I ran into the same problem recently. I'm using 1.3.3.RELEASE of spring-cloud-stream. It's said that it's already fixed in #1041 . I checked the source code and the modification has already applied to it. I can't figure out why the problem still remains in my situation.
